### PR TITLE
fix: disable api update if disabled from UI

### PR DESCRIPTION
**Pull request by rish2320**
# Description
Disable the API Update if the force security scanning is abled from UI.

Fixes https://github.com/devtron-labs/devtron/issues/2574

## How Has This Been Tested?
Tested all the using cases manually by Postman.

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [x] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

### CODE REVIEW

File: pkg/pipeline/PipelineBuilder.go [hunk 1]
```diff
@@ -1413,6 +1419,20 @@ func (impl PipelineBuilderImpl) PatchCiPipeline(request *bean.CiPatchRequest) (c
 	ciConfig.AppWorkflowId = request.AppWorkflowId
 	ciConfig.UserId = request.UserId
 	if request.CiPipeline != nil {
+		client, err := impl.K8sUtil.GetClientForInCluster()
+		if err != nil {
+			impl.logger.Errorw("exception while getting unique client id", "error", err)
+			return nil, err
+		}
+		cm, err := impl.K8sUtil.GetConfigMap(ConfigMapNamespace, DashboardConfigMap, client)
```
Comment (prakash100198) [inline pkg/pipeline/PipelineBuilder.go:1427]: handle error here

File: pkg/pipeline/PipelineBuilder.go [hunk 2]
```diff
@@ -1413,6 +1419,20 @@ func (impl PipelineBuilderImpl) PatchCiPipeline(request *bean.CiPatchRequest) (c
...
+		datamap := cm.Data
```
Comment (prakash100198) [inline pkg/pipeline/PipelineBuilder.go:1428]: What if cm is nil, handle that case also

**Review by prakash100198 — CHANGES_REQUESTED** [review]

File: pkg/pipeline/PipelineBuilder.go (hunk 2 above)
Comment (rish2320) [reply pkg/pipeline/PipelineBuilder.go:1432]: done

File: pkg/pipeline/PipelineBuilder.go (hunk 1 above)
Comment (rish2320) [reply pkg/pipeline/PipelineBuilder.go:1427]: done

File: pkg/pipeline/PipelineBuilder.go [hunk 3]
```diff
@@ -68,6 +68,9 @@ import (
 	"go.uber.org/zap"
 )
 
+const DashboardConfigMap = "dashboard-cm"
+const ConfigMapNamespace = "devtroncd"
```
Comment (vikramdevtron) [inline pkg/pipeline/PipelineBuilder.go:72]: check similar constants already used, remove namespace const.

Comment (rish2320) [reply pkg/pipeline/PipelineBuilder.go]: dashboard-cm const is not used, removed namespace const.

File: pkg/pipeline/PipelineBuilder.go (hunk 2 above)
Comment (prakash100198) [reply pkg/pipeline/PipelineBuilder.go:1431]: not able to see it handled

**Review by prakash100198 — CHANGES_REQUESTED** [review]

File: pkg/pipeline/PipelineBuilder.go [hunk 4]
```diff
@@ -1413,6 +1418,24 @@ func (impl PipelineBuilderImpl) PatchCiPipeline(request *bean.CiPatchRequest) (c
 	ciConfig.AppWorkflowId = request.AppWorkflowId
 	ciConfig.UserId = request.UserId
 	if request.CiPipeline != nil {
+		client, err := impl.K8sUtil.GetClientForInCluster()
```
Comment (vikramdevtron) [inline pkg/pipeline/PipelineBuilder.go:1421]: will it works only for in cluster ? if not handle by cluster id.

File: pkg/pipeline/PipelineBuilder.go [hunk 5]
```diff
@@ -1413,6 +1418,24 @@ func (impl PipelineBuilderImpl) PatchCiPipeline(request *bean.CiPatchRequest) (c
...
+			impl.logger.Errorw("exception while getting unique client id", "error", err)
+			return nil, err
+		}
+		cm, err := impl.K8sUtil.GetConfigMap(argo.DEVTRONCD_NAMESPACE, DashboardConfigMap, client)
+		if err != nil {
+			impl.logger.Errorw("error while getting dashboard-cm", "error", err)
+			return nil, err
+		}
+		datamap := cm.Data
+		forceScanConfig, err := strconv.ParseBool(datamap["FORCE_SECURITY_SCANNING"])
```
Comment (vikramdevtron) [inline pkg/pipeline/PipelineBuilder.go:1432]: add nil check for datamap and add constant for KEY

File: pkg/pipeline/PipelineBuilder.go (hunk 2 above)
Comment (rish2320) [reply pkg/pipeline/PipelineBuilder.go:1431]: might not be showing that time, handled it.

File: pkg/pipeline/PipelineBuilder.go (hunk 5 above)
Comment (rish2320) [reply pkg/pipeline/PipelineBuilder.go]: done

File: pkg/pipeline/PipelineBuilder.go (hunk 4 above)
Comment (rish2320) [reply pkg/pipeline/PipelineBuilder.go:1422]: Yes working .

**Review by vikramdevtron — APPROVED** [review]

**Review by vikramdevtron — DISMISSED** [review]

**Review by prakash100198 — APPROVED** [review]

**Pull request merged**